modules/nlu/src/backend/training-pipeline.ts

```diff
@@ -41,8 +39,7 @@ export type TrainInput = Readonly<{
   ctxToTrain: string[]
 }>
 
-// TODO: this should be named TrainStep and should not be returned by the pipeline
-export type TrainOutput = Readonly<{
+export type TrainStep = Readonly<{
```

Comment:
TrainStep Should not be exported. It should be used only by the training pipeline. Thats actually the reason why we are doing this refactor.